### DISCUSSION TITLE
upgrade for v1.22+ getting warnings if deprecated resources

### DISCUSSION THREAD

**Discussion by sunnythepatel** [aquasecurity/starboard#368] [Questions & Help] [answered]
I had installed starboard can anyone help me how to set starboard init with new api
 kubectl starboard init
W0128 14:15:20.519679   37541 warnings.go:67] apiextensions.k8s.io/v1beta1 CustomResourceDefinition is deprecated in v1.16+, unavailable in v1.22+; use apiextensions.k8s.io/v1 CustomResourceDefinition
W0128 14:15:20.575863   37541 warnings.go:67] apiextensions.k8s.io/v1beta1 CustomResourceDefinition is deprecated in v1.16+, unavailable in v1.22+; use apiextensions.k8s.io/v1 CustomResourceDefinition
W0128 14:15:20.582200   37541 warnings.go:67] apiextensions.k8s.io/v1beta1 CustomResourceDefinition is deprecated in v1.16+, unavailable in v1.22+; use apiextensions.k8s.io/v1 CustomResourceDefinition
W0128 14:15:20.591109   37541 warnings.go:67] apiextensions.k8s.io/v1beta1 CustomResourceDefinition is deprecated in v1.16+, unavailable in v1.22+; use apiextensions.k8s.io/v1 CustomResourceDefinition
W0128 14:15:20.598787   37541 warnings.go:67] apiextensions.k8s.io/v1beta1 CustomResourceDefinition is deprecated in v1.16+, unavailable in v1.22+; use apiextensions.k8s.io/v1 CustomResourceDefinition
W0128 14:15:20.606408   37541 warnings.go:67] apiextensions.k8s.io/v1beta1 CustomResourceDefinition is deprecated in v1.16+, unavailable in v1.22+; use apiextensions.k8s.io/v1 CustomResourceDefinition
W0128 14:15:20.612672   37541 warnings.go:67] apiextensions.k8s.io/v1beta1 CustomResourceDefinition is deprecated in v1.16+, unavailable in v1.22+; use apiextensions.k8s.io/v1 CustomResourceDefinition
W0128 14:15:20.626211   37541 warnings.go:67] apiextensions.k8s.io/v1beta1 CustomResourceDefinition is deprecated in v1.16+, unavailable in v1.22+; use apiextensions.k8s.io/v1 CustomResourceDefinition

**Comment by danielpacak** [comment, marked as answer]
This is just a deprecation notice / warning and should not have any impact of Starboard functionality run in Kubernetes cluster =< v1.21

To get rid of the warning we have to change the Go code and upgrade apiVersion from  apiextensions.k8s.io/v1beta1 to apiextensions.k8s.io/v1 in CRD files

**Reply by sunnythepatel** [reply]
Ok Thank you yeah it's working fine but just wanted to know if you are going to upgrade it

**Reply by danielpacak** [reply]
Yup. We'll do that in the upcoming release.

**Reply by danielpacak** [reply]
Unfortunately we cannot convert a discussion to an issue, but I created issue #371  manually to track discussed changes.
